### PR TITLE
Clean up after removal of runtest

### PR DESCRIPTION
**Pull request by robertodr**
Irons out some wrinkles left when removing the dependency on ``runtest``, most notably some confusion on what the working directory was supposed to be.

### CODE REVIEW

File: tests/tester.py [hunk 1]
```diff
@@ -77,37 +84,49 @@ def run(options, *, input_file, filters=None, extra_args=None):
...
                                              rtol=threshold.rtol,
                                              atol=threshold.atol)
             sys.stdout.write(f"\n{message}")
+            success = success and passed
+
+    # clean up output files on success (if requested!)
+    if success and options.clean_up:
+        computed.unlink()
+        out = Path(options.work_dir) / f"{inp_no_suffix}.out"
+        out.unlink(missing_ok=True)
```
Comment (stigrj) [inline tests/tester.py:127]: The `missing_ok` argument is not recognized. Why do we want to clean up the outputs after success? Sometimes I like to inspect the output also for successful tests.

Comment (robertodr) [reply tests/tester.py:127]: That argument is in 3.7+ only. And the removal was a workaround for not using the working directory properly.

**Comment by stigrj** [comment]
Arg!  :angry:

**Comment by robertodr** [comment]
> Arg!

Indeed

**Comment by robertodr** [comment]
It's apparently choking on this line: `out_file = f"{name}.out"` But the Docker image we are using has Python 3.6 and that is allowed syntax... :thinking:

**Comment by robertodr** [comment]
A-ha! The shebang in `mrchem.in` should be interpolated by CMake not hard-coded to `/usr/bin/env python`!

**Comment by robertodr** [comment]
I am not sure whether we want to remove `#!/usr/bin/env python` from the little `test` scripts too. That would require executing them in the test folders prepending the correct Python executable, but would avoid these arcane errors to pop up.

**Review by stigrj — APPROVED** [review]
I have no strong feelings on the last point. Do whatever you think is best and then merge.

**Comment by robertodr** [comment]
I went for `#!/usr/bin/env python3`

**Pull request merged**